### ISSUE TITLE
Unable to use my own trained models.

### ISSUE THREAD

**Issue by SirSykon** [xinntao/Real-ESRGAN#189]
Hello,

I updated to the last repo version and now I'm not able to use the models I have just trained.

Before I used the argument --input_path to indicate the trained model path. Now --input_path exists no more and I'm moving the trained models to rsgan/weights as indicates the new inference_realsrgan.py but then I get an error like the following:

Traceback (most recent call last):
  File "inference_realesrgan.py", line 128, in <module>
    main()
  File "inference_realesrgan.py", line 75, in main
    scale=netscale,
UnboundLocalError: local variable 'netscale' referenced before assignment

Any idea how should I use the weights obtained by training or fine-tuning with my own data?

**Comment by SirSykon** [comment]
It seems to work!  Thank you very much!

**Comment by theboss312** [comment]
> It seems to work! Thank you very much!

If it worked, feel free to close the issue.

**Comment by cjw19951005** [comment]
> It seems to work! Thank you very much!

I have the same problem, how to solve it？

**Comment by kodxana** [comment]
> > It seems to work! Thank you very much!
> 
> I have the same problem, how to solve it？

Just scroll up and you will see my solution :)

**Comment by cjw19951005** [comment]
> > > It seems to work! Thank you very much!
> > 
> > 
> > I have the same problem, how to solve it？
> 
> Just scroll up and you will see my solution :)

But I followed the above method and it didn't solve the problem.
I renamed the model I trained myself to RealESRGAN_x4plus.pth and the above error also occurred

**Comment by kodxana** [comment]
> > > > It seems to work! Thank you very much!
> 
> > > 
> 
> > > 
> 
> > > I have the same problem, how to solve it？
> 
> > 
> 
> > Just scroll up and you will see my solution :)
> 
> 
> 
> But I followed the above method and it didn't solve the problem.
> 
> I renamed the model I trained myself to RealESRGAN_x4plus.pth and the above error also occurred

What is your pretrain model?

**Comment by cjw19951005** [comment]
> > > > It seems to work! Thank you very much!
> > > 
> > > 
> > > I have the same problem, how to solve it？
> > 
> > 
> > Just scroll up and you will see my solution :)
> 
> But I followed the above method and it didn't solve the problem. I renamed the model I trained myself to RealESRGAN_x4plus.pth and the above error also occurred

The last file I end up training with is net_g_last.pth.
Is this file the model we need?

**Comment by kodxana** [comment]
> > > > > It seems to work! Thank you very much!
> 
> > > > 
> 
> > > > 
> 
> > > > I have the same problem, how to solve it？
> 
> > > 
> 
> > > 
> 
> > > Just scroll up and you will see my solution :)
> 
> > 
> 
> > But I followed the above method and it didn't solve the problem. I renamed the model I trained myself to RealESRGAN_x4plus.pth and the above error also occurred
> 
> 
> 
> The last file I end up training with is net_g_last.pth.
> 
> Is this file the model we need?

I mean what model you used as base when starting training.

**Comment by cjw19951005** [comment]
> > > > > It seems to work! Thank you very much!
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > I have the same problem, how to solve it？
> > 
> > 
> > > 
> > 
> > 
> > > Just scroll up and you will see my solution :)
> > 
> > 
> > But I followed the above method and it didn't solve the problem.
> > I renamed the model I trained myself to RealESRGAN_x4plus.pth and the above error also occurred
> 
> What is your pretrain model?



> > > > > > It seems to work! Thank you very much!
> > 
> > 
> > > > >
> > 
> > 
> > > > >
> > 
> > 
> > > > > I have the same problem, how to solve it？
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > Just scroll up and you will see my solution :)
> > 
> > 
> > > 
> > 
> > 
> > > But I followed the above method and it didn't solve the problem. I renamed the model I trained myself to RealESRGAN_x4plus.pth and the above error also occurred
> > 
> > 
> > The last file I end up training with is net_g_last.pth.
> > Is this file the model we need?
> 
> I mean what model you used as base when starting training.

I first did the training with total_iter: 1000000 and the pretrained model was ESRGAN_SRx4_DF2KOST_official-ff704c30.pth.

Then I did the training with total_iter:400000 and the pretrained model was net_g_1000000.pth

**Comment by knoppmyth** [comment]
The netscale error was happening to me as well. Which is odd as I used the inference tool several days ago without issue. If I specified the model_path, I'd get another error... In searching, other issues pointed back to this... However, I'm not using a trained model. I was attempting to use realesrgan-general-x4v3 (Which worked fine several days ago). The solution that worked for me, was making realesrgan-general-x4v3 the default on line 21 of the inference script. Hopefully this works for others as well. #398 #342 #242
